public/app/features/explore/correlationEditLogic.test.ts

```diff
@@ -0,0 +1,39 @@
+import { CORRELATION_EDITOR_POST_CONFIRM_ACTION } from 'app/types';
+
+import { showModalMessage } from './correlationEditLogic';
+
+// note, closing the editor does not care if isLeft is true or not. Both are covered for regression purposes.
+describe('correlationEditLogic', function () {
+  it.each`
+    action                                                      | isLeft   | dirCor   | dirQuer  | expected
+    ${CORRELATION_EDITOR_POST_CONFIRM_ACTION.CLOSE_PANE}        | ${false} | ${false} | ${false} | ${undefined}
+    ${CORRELATION_EDITOR_POST_CONFIRM_ACTION.CLOSE_PANE}        | ${false} | ${true}  | ${false} | ${'Closing the pane will cause the correlation in progress to be lost. Would you like to save before continuing?'}
+    ${CORRELATION_EDITOR_POST_CONFIRM_ACTION.CLOSE_PANE}        | ${false} | ${false} | ${true}  | ${'Closing the pane will lose the changed query. Would you like to save before continuing?'}
```

Comment:
The modal seems to be shown twice when closing the pane:


https://github.com/grafana/grafana/assets/745532/581dd28f-ed71-4ace-8d18-a886cff18aa8

Reply:
Fixed!